libc/hdr/float_macros.h

```diff
@@ -0,0 +1,19 @@
+//===-- Definition of macros from math.h ----------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_LIBC_HDR_FLOAT_MACROS_H
+#define LLVM_LIBC_HDR_FLOAT_MACROS_H
+
+#ifndef LIBC_FULL_BUILD
+// Overlay mode
+#include <float.h>
+#endif // !LLVM_LIBC_FULL_BUILD
```

Comment:
This pattern feels different than the rest of our hdr/ dir.  Usually, the pattern is:

```c
#ifdef LIBC_FULL_BUILD
#include our macros
#else
#include system header
#endif
```

Reply:
Changed back to the common pattern of `hdr` dir.